### PR TITLE
[MIFOSX-2114] Change assignment staff to groups and clients in center.

### PR DESCRIPTION
**Pull request by k4wel**
Recreated from #275.

### CODE REVIEW

**Comment by vorburger** [comment]
@TypicalGitHubUser would you mind to re-rebase this, so that Travis CI will run on it?

@nazeer1100126 will you re-review this as requested in #275 ?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java [hunk 1]
```diff
@@ -247,9 +247,15 @@ public CommandProcessingResult createClient(final JsonCommand command) {
...
             Staff staff = null;
-            final Long staffId = command.longValueOfParameterNamed(ClientApiConstants.staffIdParamName);
-            if (staffId != null) {
-                staff = this.staffRepository.findByOfficeHierarchyWithNotFoundDetection(staffId, clientOffice.getHierarchy());
+            if (clientParentGroup != null) {
+                if (clientParentGroup.getParent() != null) {
+                    staff = clientParentGroup.getParent().getStaff();
+                }
+            } else {
+                final Long staffId = command.longValueOfParameterNamed(ClientApiConstants.staffIdParamName);
```
Comment (vishwasbabu) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java:255]: If the Client has a parent group, but the parent group does not have a center, the else condition would not execute and staffId passed in through the API call would be lost. You would now end up with a Client with no staff assigned when the API caller had explicitly passed in a staffId

Ideally, the behavior should be such that.

1) If the staffId is passed in through the API call, set the same to the client
2) If not, inherit the staffId from the client's parent (if the client has a parent)

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java [hunk 2]
```diff
@@ -158,9 +158,15 @@ private CommandProcessingResult createGroupingType(final JsonCommand command, fi
             validateOfficeOpeningDateisAfterGroupOrCenterOpeningDate(groupOffice, groupLevel, activationDate);
 
             Staff staff = null;
-            final Long staffId = command.longValueOfParameterNamed(GroupingTypesApiConstants.staffIdParamName);
-            if (staffId != null) {
-                staff = this.staffRepository.findByOfficeHierarchyWithNotFoundDetection(staffId, groupOffice.getHierarchy());
+            if ((GroupTypes.GROUP).equals(groupingType)) {
+                if (parentGroup != null) {
+                    staff = parentGroup.getStaff();
```
Comment (vishwasbabu) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java:163]: Same issue as the previous comment

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java [hunk 3]
```diff
@@ -388,11 +394,16 @@ private CommandProcessingResult updateGroupingType(final Long groupId, final Jso
...
 
             if (actualChanges.containsKey(GroupingTypesApiConstants.staffIdParamName)) {
-                final Long newValue = command.longValueOfParameterNamed(GroupingTypesApiConstants.staffIdParamName);
-
                 Staff newStaff = null;
-                if (newValue != null) {
-                    newStaff = this.staffRepository.findByOfficeHierarchyWithNotFoundDetection(newValue, groupHierarchy);
+                if ((GroupTypes.GROUP).equals(groupingType)) {
+                    if (groupForUpdate.getParent() != null) {
+                        newStaff = groupForUpdate.getParent().getStaff();
```
Comment (vishwasbabu) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java:400]: Here too

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java (hunk 1 above)
Comment (vishwasbabu) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java:255]: Please add an integration test case which validates the functional changes made

**Review by vishwasbabu — CHANGES_REQUESTED** [review]

**Comment by vishwasbabu** [comment]
@TypicalGitHubUser : Please let us know if you have the time to address these review comments anytime in the near future. If not, we shall be closing this PR on the 15th of March in an effort to help committers have a clearer view of pending pull requests. Hope you understand and think this is fair.

**Comment by k4wel** [comment]
@vishwasbabu No, I won't be able to do this in near future.

**Comment by TypicalGitHubUser** [comment]
@vishwasbabu No, I won't be able to do this in near future.

**Comment by vishwasbabu** [comment]
Closing this PR

**Pull request closed without merging**